redbot/cogs/mutes/mutes.py

```diff
@@ -1559,6 +1559,9 @@ async def mute_user(
             try:
                 await user.add_roles(role, reason=reason)
                 await self.config.guild(guild).muted_users.set(self._server_mutes[guild.id])
+            except discord.errors.HTTPException:
```

Comment:
it might be worth adding the guild.muted_users.set here to ensure the user is part of the stored _server_mutes